tests/core/test_integration.py

```diff
@@ -1651,6 +1651,572 @@ def test_restatement_plan_ignores_changes(init_and_plan_context: t.Callable):
     context.apply(plan)
 
 
+def test_restatement_plan_hourly_with_downstream_daily_restates_correct_intervals(tmp_path: Path):
```

Comment:
This test is for an existing bug where restating in the same environment still doesn't work properly if the interval being restated is smaller than the interval_unit of one of the models